build-samples/build_samples.sh

```diff
@@ -15,6 +15,23 @@ if [ $? -eq 0 ]; then
 	export LLVM="-$LLVM_VER"
 fi
 
+if [[ $(uname -m) != "$TARGETARCH" ]]; then
+	# Cross-compiling
+	linuxarch="$TARGETARCH"
+	case "$TARGETARCH" in
+	riscv64)
+		linuxarch="riscv"
+		;;
+	aarch64)
+		linuxarch="arm64"
+		;;
+	*)
+		;;
+	esac
+	export ARCH="$linuxarch"
+	export CROSS_COMPILE="$TARGETARCH-linux-gnu-"
+fi
+
```

Comment:
This is probably worth putting in helpers.sh so it can be re-used across the different actions uniformally.